### PR TITLE
Feature/localhost

### PR DESCRIPTION
**Pull request by reftel**
Please ensure your pull request adheres to the following guidelines: 

- [X] Use the following formatting style: [SonarSource/sonar-developer-toolset](https://github.com/SonarSource/sonar-developer-toolset#code-style)
- [X] Unit tests are passing and you provided a unit test for your fix
- [X] ITs should pass : To run ITs locally, checkout the [README](https://github.com/SonarSource/sonar-java/blob/master/README.md) of the project.
- [N/A] If there is a [Jira](http://jira.sonarsource.com/browse/SONARJAVA) ticket available, please make your commits and pull request start with the ticket number (SONARJAVA-XXXX)

### CODE REVIEW

File: java-checks/src/main/java/org/sonar/java/checks/HardcodedIpCheck.java [hunk 1]
```diff
@@ -20,20 +20,29 @@
...
 import org.sonar.plugins.java.api.tree.Tree;
 
-import java.util.regex.Matcher;
-import java.util.regex.Pattern;
-
 @Rule(key = "S1313")
 public class HardcodedIpCheck extends BaseTreeVisitor implements JavaFileScanner {
 
   private static final Matcher IP = Pattern.compile("[^\\d.]*?((?:\\d{1,3}\\.){3}\\d{1,3}(?!\\d|\\.)).*?").matcher("");
+  private static final List<List<Integer>> WHITELIST = new ArrayList<List<Integer>>();
```
Comment (sonartech-externalpr) [inline java-checks/src/main/java/org/sonar/java/checks/HardcodedIpCheck.java]: MINOR Replace the type specification in this constructor call with the diamond operator ("<>"). [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS2293)

**Comment by sonartech-externalpr** [comment]
SonarQube analysis reported 2 issues
* MINOR 2 minor

Watch the comments in this conversation to review them.

**Comment by m-g-sonar** [comment]
Hello, 

Thanks for your contribution. Now, before considering it:
- Even if I understand your needs, can you please open a thread in our Mailing List introducing your PR (by sending an email to sonarqube[at]googlegroups.com, or directly through https://groups.google.com/forum/#!forum/sonarqube). This step would allow to include everyone involved in rule implementation in the process.
- Can you please fix the result of the ruling in order to take into account the differences you introduced?

Regards,
Michael

File: java-checks/src/main/java/org/sonar/java/checks/HardcodedIpCheck.java [hunk 2]
```diff
@@ -48,21 +57,28 @@ public void visitLiteral(LiteralTree tree) {
     if (tree.is(Tree.Kind.STRING_LITERAL)) {
       IP.reset(tree.value());
       if (IP.matches()) {
+        ArrayList<Integer> octets = new ArrayList<Integer>(4);
```
Comment (sonartech-externalpr) [inline java-checks/src/main/java/org/sonar/java/checks/HardcodedIpCheck.java]: MINOR Replace the type specification in this constructor call with the diamond operator ("<>"). [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS2293)

**Comment by sonartech-externalpr** [comment]
SonarQube analysis reported 2 issues
* MINOR 2 minor

Watch the comments in this conversation to review them.

**Comment by reftel** [comment]
Opened thread: https://groups.google.com/forum/#!topic/sonarqube/36EofymqRxI

**Comment by m-g-sonar** [comment]
Hey @reftel , thanks for raising the point on our mailing list, it triggered discussions that maybe would not have happened when only discussing on a PR thread.

For reasons mentioned in the Mailing list thread, we are not going to merge your PR. As being a security rule, it sounds more important to us to ask users to carefully study each cases and flag issue as being 'FP' or 'Won't Fix' when the rule does not apply.

Cheers,
Michael

**Pull request closed without merging**